### ISSUE TITLE
when using vf-install.php, elite_schema table isn't created.

### ISSUE THREAD

**Issue by kylecannon** [vehiclefits/vfmagento#23]
The following tables are created when running it:

elite_0_definition
elite_0_mapping
elite_level_0_engine
elite_level_0_make
elite_level_0_model
elite_level_0_year

the vf-install.php tells me that it ran successfully as well.

**Comment by kylecannon** [comment]
Running just the generate database sql, I get the following output:

``` sql
-- Schema for --
CREATE TABLE IF NOT EXISTS `elite_level_0_year` (
`id` int(255) NOT NULL AUTO_INCREMENT,
`title` varchar(255) NOT NULL,
PRIMARY KEY (`id`),
KEY `title` (`title`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
CREATE TABLE IF NOT EXISTS `elite_level_0_make` (
`id` int(255) NOT NULL AUTO_INCREMENT,
`title` varchar(255) NOT NULL,
PRIMARY KEY (`id`),
KEY `title` (`title`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
CREATE TABLE IF NOT EXISTS `elite_level_0_model` (
`id` int(255) NOT NULL AUTO_INCREMENT,
`title` varchar(255) NOT NULL,
PRIMARY KEY (`id`),
KEY `title` (`title`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
CREATE TABLE IF NOT EXISTS `elite_level_0_engine` (
`id` int(255) NOT NULL AUTO_INCREMENT,
`title` varchar(255) NOT NULL,
PRIMARY KEY (`id`),
KEY `title` (`title`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
CREATE TABLE IF NOT EXISTS `elite_0_mapping` (`id` int(50) NOT NULL AUTO_INCREMENT,`year_id` int(15) NOT NULL,`make_id` int(15) NOT NULL,`model_id` int(15) NOT NULL,`engine_id` int(15) NOT NULL,`entity_id` int(25) NOT NULL,`universal` int(1) NOT NULL COMMENT 'if there is a row with this flag set for a product ( entity_id ) then it should be returned universally for all vehicles',`related` int(1) NOT NULL,PRIMARY KEY (`id`),KEY `universal` (`universal`),KEY `year_id` (`year_id`),KEY `make_id` (`make_id`),KEY `model_id` (`model_id`),KEY `engine_id` (`engine_id`)) ENGINE=InnoDB CHARSET=utf8;ALTER TABLE `elite_0_mapping` ADD `price` FLOAT NOT NULL ;ALTER TABLE `elite_0_mapping` ADD INDEX ( `entity_id` ) ;ALTER TABLE `elite_0_mapping` ADD UNIQUE ( `year_id`,`make_id`,`model_id`,`engine_id`,universal,entity_id );CREATE TABLE IF NOT EXISTS `elite_0_definition` (`id` int(50) NOT NULL AUTO_INCREMENT,`year_id` int(15) NOT NULL,`make_id` int(15) NOT NULL,`model_id` int(15) NOT NULL,`engine_id` int(15) NOT NULL,PRIMARY KEY (`id`),KEY `year_id` (`year_id`),KEY `make_id` (`make_id`),KEY `model_id` (`model_id`),KEY `engine_id` (`engine_id`)) ENGINE=InnoDB CHARSET=utf8;; ALTER TABLE `elite_0_definition` ADD UNIQUE ( `year_id`,`make_id`,`model_id`,`engine_id` );
        ALTER TABLE `sales_flat_order_item` ADD `elite_paint` INT( 25 ) NOT NULL ;
        ALTER TABLE `sales_flat_order_item` ADD `elite_paint_other` VARCHAR( 25 ) NOT NULL ;
        ALTER TABLE `sales_flat_quote_item` ADD `elite_paint` INT( 25 ) NOT NULL ;
        ALTER TABLE `sales_flat_quote_item` ADD `elite_paint_other` VARCHAR( 25 ) NOT NULL ;
```

**Comment by kylecannon** [comment]
Confirmed that if i run `[vagrant@dev Elite]$ php tasks/schema.php --force --levels="year,make,model"` it works and creates the following tables:

```
mysql> show tables LIKE 'elite%';
+----------------------------------------+
| Tables_in_vehiclefits_magento (elite%) |
+----------------------------------------+
| elite_1_definition                     |
| elite_1_mapping                        |
| elite_definition_wheel                 |
| elite_import                           |
| elite_level_1_make                     |
| elite_level_1_model                    |
| elite_level_1_year                     |
| elite_mapping_notes                    |
| elite_mapping_paint                    |
| elite_note                             |
| elite_product_servicecode              |
| elite_product_tire                     |
| elite_product_wheel                    |
| elite_product_wheeladapter             |
| elite_schema                           |
| elite_vehicle_tire                     |
| elite_version                          |
+----------------------------------------+
17 rows in set (0.00 sec)
```

However, `vf-install.php` doesn't work correctly.

**Comment by joshribakoff** [comment]
I created a new label called task, it felt awkward calling this a bug when it doesn't affect released versions of the software.. its more like a reminder that we have a half finished feature

**Comment by kylecannon** [comment]
You should note in the install instructions to not use the vf-install then and use the schema task

**Comment by joshribakoff** [comment]
I'm going to fix the issue so no need for that, I just want to distinguish between an issue that prevents the mainline from being deployable, and an issue that affects a deployed version of the software.  I definitely agree there is a missing test & this should be fixed, and technically our build is "broken" even though the automated tests pass, in the sense we cannot deploy until it is fixed (even though that fix is extremely trivial). My reason I want to label it something other than 'bug' is I don't want people to look at the issue tracker & think its a bug affecting a released version of the software, although you are right its a bug in the sense I failed at keeping the mainline deployable at all times.

**Comment by joshribakoff** [comment]
Moving this issue to the core repo's issue tracker

**Comment by joshribakoff** [comment]
Or, I don't know... I'm not crazy about these multiple repos, or maybe my issue is with Git's issue tracker. The fix to this was a commit in the core repo. Seems like unnecessary overhead to try to figure out which repo an issue "technically" belongs to and there's no way to reassign an issue to another repo even if we agreed on which repo it belongs in. Anyways its fixed.

**Comment by drmmr763** [comment]
Hey all:

I'm just running into this issue now and I'm wondering what the status of this issue. I've got this issue where Magento is telling me that there are tables missing here. I checked for the schema/tasks.php file referenced above but I didn't find it. 

Also I see there is a composer.json file -- do I need to run composer? I didn't see any mention of that in documentation. I did run the vf-install.php file, but that was it. Did I miss something?

Thanks for any help here. We are undertaking a rather large-scale project that we intend to use vehiclefits on.
